index.js

```diff
@@ -88,6 +92,19 @@ function fastifyOauth2 (fastify, options, next) {
     schema = { tags }
   } = options
 
+  const userAgent = options.userAgent
+    ? `${options.userAgent} ${USER_AGENT}`
```

Comment:
So you can not truely override the useragent?

Reply:
Expected more something like

```js
  let userAgent = {}
  switch (options.userAgent) {
    case false:
      break
    case undefined:
      userAgent['User-Agent'] = USER_AGENT
      break
    default:
      userAgent['User-Agent'] = USER_AGENT
      break
  }

  const credentials = {
    ...options.credentials,
    http: {
      ...options.credentials.http,
      headers: {
        ...userAgent,
        ...options.credentials.http?.headers
      }
    }
  }
```

Reply:
> So you can not truely override the useragent?

You can, if you override the http settings

On other comment:

I prefer using `const` over `let` and avoid `switch` unless checking one bar against multiple values (as a missing `break` can cause havoc and it’s a bit unusual in that body of a `case` doesn’t come with its own scope, unless you add one, so it’s easy for people to make mistakes with)

Reply:
If I set userAgent to "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0)", then the plugin is changing the useragent to: "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0) fastify/oauth2"

So how can I override the useragent?

Reply:
@Uzlopak This is what the RFC 9110 says, my emphasis:

> The User-Agent field value consists of **one or more product identifiers**, each followed by zero or more comments (Section 5.6.5), which together identify the user agent software and its significant subproducts. By convention, **the product identifiers are listed in decreasing order of their significance for identifying the user agent software**. Each product identifier consists of a name and optional version.

This is what this PR up to now has done + I have added tests + comments on how to override it if one wants to.

But to avoid confusion I have now removed such prepending of the additional user-agent and instead outright replaces the default one. Removes possibly valuable information but I hope that fixes your concerns.